pkg/vmprovider/providers/vsphere/vmprovider_vm_utils.go

```diff
@@ -37,6 +37,9 @@ func GetVirtualMachineClass(
 	k8sClient ctrlclient.Client) (*vmopv1.VirtualMachineClass, error) {
 
 	className := vmCtx.VM.Spec.ClassName
+	if lib.IsNamespacedClassAndWindowsFSSEnabled() {
```

Comment:
You can dedupe some of this and the following block so that you don't need the separate `resolveVMClass` func?

```golang
key := client.ObjectKey{Name: className}

if lib.IsNamespacedClassAndWindowsFSSEnabled() {
        // namespace scoped VM classes
        key.Namespace = ctx.VM.Namespace
}
// and then go on to fetch the class
vmClass := &vmopv1.VirtualMachineClass{}
if err := k8sClient.Get(vmCtx, ctrlclient.ObjectKey{Name: className}, vmClass); err != nil {
        msg := fmt.Sprintf("Failed to get VirtualMachineClass: %s", className)
	conditions.MarkFalse(vmCtx.VM,
	        vmopv1.VirtualMachinePrereqReadyCondition,
	        vmopv1.VirtualMachineClassNotFoundReason,
	        vmopv1.ConditionSeverityError,
	        msg)
        return nil, errors.Wrap(err, msg)
}

Reply:
I have thought about this before, but since we need to return immediately after getting the VM Class when the FSS is enabled, I feel like have a separate function may be more readable? Otherwise, we need to check the FSS twice, one for the key, another for return immediately check.